domain/model/service/modelservice.go

```diff
@@ -81,18 +80,13 @@ type AgentBinaryFinder interface {
 	HasBinariesForVersion(version.Number) (bool, error)
 }
 
-// ResourceCreationProvider is an subset of the [environs.Environ] interface that is used for the model service
-// to create resources for a new model.
-type ResourceCreationProvider interface {
-	// Create creates the environment for a new hosted model.
-	//
-	// This will be called before any workers begin operating on the
-	// Environ, to give an Environ a chance to perform operations that
-	// are required for further use.
-	//
-	// Create is not called for the initial controller model; it is
-	// the Bootstrap method's job to create the controller model.
-	Create(environsContext.ProviderCallContext, environs.CreateParams) error
+// ModelResourcesProvider mirrors the [environs.ModelResources] interface that is
+// used by the model service when creating a new model.
+type ModelResourcesProvider interface {
+	// ValidateModelCreation is part of the [environs.ModelResources] interface.
+	ValidateModelCreation(ctx context.Context) error
```

Comment:
This is named poorlThis is named poorly in my opinion. Model's aren't created in provides they are just side affected. The wording of this leaves me wondering what is actually doing.y in my opinion. Model's aren't created in provides they are just side affected. The wording of this leaves me wondering what is actually doing.